openwisp_controller/subnet_division/tasks.py

```diff
@@ -116,12 +117,35 @@ def _create_ipaddress_and_subnetdivision_index_objects(ips, indexes):
 
 
 @shared_task
-def provision_subnet_ip_for_existing_devices(organization_id):
-    for config in (
-        Config.objects.select_related('device', 'device__organization')
-        .filter(device__organization_id=organization_id)
-        .iterator()
-    ):
-        DeviceSubnetDivisionRuleType.provision_receiver(
-            config, created=True,
+def provision_subnet_ip_for_existing_devices(rule_id):
+    try:
+        rule = SubnetDivisionRule.objects.get(id=rule_id)
+    except SubnetDivisionRule.DoesNotExist as error:
+        logger.warning(
+            'Failed to provision IPs on existing devices for Subnet '
+            f'Division Rule with id: "{rule_id}", reason: {error}'
+        )
+        return
+    if 'device' in rule.type:
+        for config in (
+            Config.objects.select_related('device', 'device__organization')
+            .filter(device__organization_id=rule.organization_id)
+            .iterator()
+        ):
+            rule.rule_class.provision_receiver(
+                config, created=True,
+            )
+    elif 'vpn' in rule.type:
+        organization_filter = Q(organization_id=rule.organization_id) | Q(
+            organization_id=None
+        )
+        vpn_qs = (
+            Vpn.objects.filter(subnet=rule.master_subnet)
+            .filter(organization_filter)
+            .values_list('id')
+        )
+        qs = VpnClient.objects.filter(
+            vpn__in=vpn_qs, config__device__organization_id=rule.organization_id
         )
+        for vpn_client in qs:
+            rule.rule_class.provision_receiver(instance=vpn_client, created=True)
```

Comment:
Why keep all this logic here when this is clearly specific for each rule?
This if/else is a poor approach, considering the effort we've done to implement this as an easily extensible feature, I believe we should be consistent and move this logic to a dedicated method on each rule class.

Eg:

```
rule.rule_class.provision_existing_devices()
```

What has been preventing you from doing so?